primitives/src/parachain.rs

```diff
@@ -78,6 +78,67 @@ pub type ValidatorPair = validator_app::Pair;
 /// so we define it to be the same type as `SessionKey`. In the future it may have different crypto.
 pub type ValidatorSignature = validator_app::Signature;
 
+/// Retriability for a given active para.
+#[derive(Clone, Eq, PartialEq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Debug))]
+pub enum Retriable {
+	/// Ineligible for retry. This means it's either a parachain which is always scheduled anyway or
+	/// has been removed/swapped.
+	Never,
+	/// Eligible for retry; the associated value is the number of retries that the para already had.
+	WithRetries(u32),
+}
+
+/// Type determining the active set of parachains in current block.
+pub trait ActiveParas {
```

Comment:
```suggestion
pub trait ActiveParachains {
```

typically we avoid abbreviation (`use primitives::parachain::Id as ParaId` is the only exception, because it is so pervasive)

Reply:
I've tended to use "paras" to refer to both parachains and parathreads; in so far as the code that this PR touches, i've tried to reserve "parachains" to specifically those which are (semi-)permanent.